ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixParticipant.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ */
+package com.github.ambry.clustermap;
+
+import com.github.ambry.config.ClusterMapConfig;
+import java.io.IOException;
+import org.apache.helix.HelixManager;
+import org.apache.helix.InstanceType;
+import org.apache.helix.model.LeaderStandbySMD;
+import org.apache.helix.participant.StateMachineEngine;
+import org.json.JSONException;
+
+
+/**
+ * An implementation of {@link ClusterParticipant} that registers registers as a participant to a Helix cluster.
+ */
+class HelixParticipant implements ClusterParticipant {
+  private final String clusterName;
+  private final String zkConnectStr;
+  private final HelixFactory helixFactory;
+  private HelixManager manager;
+
+  /**
+   * Instantiate a HelixParticipant.
+   * @param clusterMapConfig the {@link ClusterMapConfig} associated with this participant.
+   * @param helixFactory the {@link HelixFactory} to use to get the {@link HelixManager}.
+   * @throws JSONException if there is an error in parsing the JSON serialized ZK connect string config.
+   */
+  HelixParticipant(ClusterMapConfig clusterMapConfig, HelixFactory helixFactory) throws IOException {
+    clusterName = clusterMapConfig.clusterMapClusterName;
+    this.helixFactory = helixFactory;
+    if (clusterName.isEmpty()) {
+      throw new IllegalStateException("Clustername is empty in clusterMapConfig");
+    }
+    try {
+      zkConnectStr = ClusterMapUtils.parseZkJsonAndPopulateZkInfo(clusterMapConfig.clusterMapDcsZkConnectStrings)
```

Comment:
Probably. Have added this to the backlog to see if we can nicely pass this for the `HelixParticipant` and the entire map for the `HelixClusterManager`. There are cascading changes (different exceptions thrown that have to be either caught or thrown with an update to the interface method signature).